tests/unit/job/updateJobHandler/updateJobHandler.spec.ts

```diff
@@ -81,10 +81,39 @@ describe('updateJobHandler', () => {
     });
   });
   describe('handleJobFinalize', () => {
-    it('should throw not implemented Error', async () => {
-      const { updateJobHandler } = setupUpdateJobHandlerTest();
+    it('should handle job finalize successfully', async () => {
+      const { updateJobHandler, catalogClientMock, jobManagerClientMock, queueClientMock } = setupUpdateJobHandlerTest();
+      const job = ingestionUpdateJob;
+      const task = finalizeTaskForIngestionUpdate;
+
+      catalogClientMock.update.mockResolvedValue(undefined);
+      jobManagerClientMock.updateTask.mockResolvedValue(undefined);
+
+      await updateJobHandler.handleJobFinalize(job, task);
+
+      expect(jobManagerClientMock.updateJob).toHaveBeenCalledWith(job.id, {
+        status: OperationStatus.COMPLETED,
+        reason: 'Job completed successfully',
+      });
+      expect(catalogClientMock.update).toHaveBeenCalledWith(job);
+      expect(jobManagerClientMock.updateTask).toHaveBeenCalledWith(job.id, task.id, { parameters: { updatedInCatalog: true } });
+      expect(queueClientMock.ack).toHaveBeenCalledWith(job.id, task.id);
```

Comment:
Please add `updateJob` toHaveBeenCalledTimes(1)

Reply:
already exists (changed hierarchy for clarity )